### PR TITLE
fix: add nil/len guards for RunCommand

### PR DESCRIPTION
**Pull request by shibumi**
*Please fill in the fields below to submit a pull request.  The more
information that is provided, the better.*

**Fixes issue #**: #124

**Description of pull request**:

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by shibumi** [comment]
This PR is missing more tests for the InTotoRun function. Right now I am only testing the RunCommand func. We might want to test link creation for an empty CmdArgs parameter as well

File: in_toto/runlib_test.go [hunk 1]
```diff
@@ -360,6 +360,23 @@ func TestRunCommand(t *testing.T) {
 	}
 }
 
+func TestRunCommandErrors(t *testing.T) {
```
Comment (adityasaky) [inline in_toto/runlib_test.go:363]: Can / should we also test by invoking `InTotoRun`?

Comment (adityasaky) [reply in_toto/runlib_test.go:363]: Oh whoops, I missed your comment on the PR.

Comment (shibumi) [reply in_toto/runlib_test.go:363]: yes we can. I will add a few more tests for this.

**Comment by shibumi** [comment]
> We should add -x / --no-command to the CLI to indicate no commands.

Is it even possible to create an in-toto link attestation without a command being run?

**Comment by adityasaky** [comment]
Yes, the field should just be empty. The reference implementation supports it. Also note that links generates via record always have the command field empty.

**Comment by shibumi** [comment]
Just to be sure:
> We should add -x / --no-command to the CLI to indicate no commands.

The `-x` flag is for the subcommand `run`, right?

File: in_toto/runlib_test.go (hunk 1 above)
Comment (adityasaky) [reply in_toto/runlib_test.go:380]: Reviving this one. I see what you mean but I think it's okay to test that the no command flag works as intended and still results in a link, no?

Comment (adityasaky) [reply in_toto/runlib_test.go:380]: @shibumi added a test checking if link metadata is successfully generated. Also caught that a minor difference in `byProducts` had crept in.

**Comment by adityasaky** [comment]
@shibumi @pxp928 I think this is good to go, please take a look. If we merge this in, we can soon cut a release I think.

**Review by pxp928 — APPROVED** [review]
Looks good based on the code.

**Pull request merged**